src/System.Windows.Extensions/src/System/Security/Cryptography/X509Certificates/X509Utils.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal static class X509Utils
+    {
+        internal const uint CERT_STORE_ENUM_ARCHIVED_FLAG = 0x00000200;
+        internal const uint CERT_STORE_CREATE_NEW_FLAG = 0x00002000;
+
+        internal static SafeCertContextHandle GetCertContext(X509Certificate2 certificate)
```

Comment:
This is called "GetCertContext", but it allocates a new SafeHandle, which needs to be disposed (and none of the callers seem to).  Maybe a better name can be come up with.

And making all of the callers of this dispose their handles at the appropriate time seems good, too.